nio/client/async_client.py

```diff
@@ -1007,3 +1011,144 @@ def add_response_callback(
             path,
             response_data=(room_id, )
         )
+
+    @logged_in
+    async def get_profile(self, user_id=None):
+        # type: (Optional[str]) -> Union[ProfileGetResponse, ProfileGetError]
+        """Get an user's combined profile information.
+
+        This queries the display name and avatar matrix content URI of an user
+        from the server. Additional profile information may be present.
+        The currently logged in user is queried if no user is specified.
+
+        Returns either a `ProfileGetResponse` if the request was
+        successful or a `ProfileGetError` if there was an error
+        with the request.
+
+        Args:
+            user_id (str): User id of the user to get the profile for.
+        """
+        method, path = Api.profile_get(
+            self.access_token,
+            user_id or self.user_id
+        )
+
+        return await self._send(
+            ProfileGetResponse,
+            method,
+            path,
+        )
+
+    @logged_in
+    async def get_displayname(
+            self,
+            user_id=None  # type: Optional[str]
+    ):
+        # type: (...) -> Union[ProfileGetDisplayNameResponse, ProfileGetDisplayNameError]
+        """Get an user's display name.
+
+        This queries the display name of an user from the server.
+        The currently logged in user is queried if no user is specified.
+
+        Returns either a `ProfileGetDisplayNameResponse` if the request was
+        successful or a `ProfileGetDisplayNameError` if there was an error
+        with the request.
+
+        Args:
+            user_id (str): User id of the user to get the display name for.
+        """
+        method, path = Api.profile_get_displayname(
+            self.access_token,
+            user_id or self.user_id
+        )
+
+        return await self._send(
+            ProfileGetDisplayNameResponse,
+            method,
+            path,
+        )
+
+    @logged_in
+    async def set_displayname(self, displayname):
+        # type: (str) -> Union[ProfileSetDisplayNameResponse, ProfileSetDisplayNameError]
```

Comment:
The error class isn't imported.

If the type ends up getting too long you can make a type alias:
```python
_ProfileSetDisplayNameT = Union[
    ProfileSetDisplayNameResponse,
    ProfileSetDisplayNameError
]
```
And put _ProfileSetDisplayNameT after the ->.